selene/src/main.rs

```diff
@@ -391,6 +391,55 @@ fn read_file(checker: &Checker<toml::value::Value>, filename: &Path) {
     );
 }
 
+/// Reads a config file and return content and path. None if not found or empty.
+fn read_config_file() -> Option<(String, Option<PathBuf>)> {
+    let config_paths_to_check = [".selene.toml", "selene.toml"];
+    let mut config_contents = String::new();
+    let mut config_path = None;
+
+    for path in &config_paths_to_check {
+        if let Ok(contents) = fs::read_to_string(path) {
+            config_contents = contents;
+            config_path = Some(PathBuf::from(path));
+            break;
+        }

```

Comment:
You're discarding `fs::read_to_string` error messages here. Returning the function as an err result would make more sense than option.

Reply:
We silence that error because if no file has been found, we return None. That way we can treat the lack of config file it as we decide for each case inside `parse_config_file_as_string()`, or `parse_config_file_as_checkerconfig()`.

```rust
    for path in &CONFIG_PATHS_TO_CHECK {
        if let Ok(contents) = fs::read_to_string(path) {
            config_contents = contents;
            config_path = Some(PathBuf::from(path));
            break;
        }
    }

    if config_contents.is_empty() {
        return None;
    }

    Some((config_contents, config_path))
```

Reply:
This prevents other errors from propagating, like trying to read invalid UTF-8 characters. I’d check if the path exists first so you can handle that separately.

Reply:
fixed in https://github.com/Kampfkarren/selene/pull/591/commits/ade463c78eca904e2d3fd801c3f6a792f258edc6